prov/efa/src/rxr/rxr_pkt_cmd.c

```diff
@@ -536,12 +552,40 @@ void rxr_pkt_handle_data_copied(struct rxr_ep *ep,
 				size_t data_size)
 {
 	struct rxr_rx_entry *rx_entry;
+	ssize_t ret;
 
 	rx_entry = pkt_entry->x_entry;
 	assert(rx_entry);
 	rx_entry->bytes_copied += data_size;
 
 	if (rx_entry->total_len == rx_entry->bytes_copied) {
+		/*
+		 * rxr_cq_handle_rx_completion() releases pkt_entry, thus
+		 * we do not release it here.
+		 */
+		if (rxr_get_base_hdr(pkt_entry->pkt)->flags & RXR_REQ_WAIT_RECEIPT) {
+		    rx_entry->tx_id = rxr_get_dc_rtm_base_hdr(pkt_entry->pkt)->tx_id;
+			rx_entry->msg_id = rxr_get_rtm_base_hdr(pkt_entry->pkt)->msg_id;
+			rx_entry->addr = pkt_entry->addr;
+			ret = rxr_pkt_post_ctrl_or_queue(ep,
+							 RXR_RX_ENTRY,
+							 rx_entry,
+							 RXR_RECEIPT_PKT, 0);
+			rxr_cq_handle_rx_completion(ep, pkt_entry, rx_entry);
+			rxr_msg_multi_recv_free_posted_entry(ep, rx_entry);
```

Comment:
where is rx_entry released? We do not want to release it here, because post of receipt might fail because of -FI_EAGAIN, in that case when we repost, we will need rx_entry. I think the write place to release rx_entry is in ```rxr_pkt_handle_receipt_sent```

Reply:
Yeah. They should be only released if `rxr_pkt_post_ctrl_or_queue` is called successfully on a `RXR_RECEIPT_PKT`. I will resolve that.

Reply:
rx_entries are all released in `rxr_pkt_handle_receipt_sent` right now, mark as resolved.